src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,65 @@
+package duke.command;
+
+import duke.exception.DukeException;
+
+/**
+ * Deals with making sense of the user input.
+ */
+public class Parser {
+
+    /**
+     * Deals with making sense of the user input.
+     * Returns a command object after decoding the user input.
+     * Throws exception if user input is not a recognizable command.
+     *
+     * @param input Contains the command name or other information to create command object.
+     * @return The appropriate command object that can be executed.
+     */
+    public static Command parse(String input) throws DukeException {
```

Comment:
could maybe implement java enums with the switch/case method for all the different types of commands instead of using if/else statements